Feature/calculator model

hihiboss opened the pull request:
## Related Issue

resolve: #32

## Description

- `measuring tool list` 에 대한 getter 리팩토링
- 재료(`meterial`) 에 대한 모델 및 매니저 구현
- 계산기 모델 (`calculator`) 구현
- `CalculatorViewController`에 계산기 모델 DI 구현

## Calculator Model 사용법
`CalculatorViewController`에 `calculator` 모델이 이미 DI 되어있음!
(`CalculatorViewController.swift` 66번째 줄과 `SwinjectStoryboard+Setup.swift` 44~48번째 줄 참고)

```swift
    // 재료 모드
    func calculate(srcAmount: String, srcUnit: MeasuringTool, dstTool: MeasuringTool) -> String

    // 재료 모드 (재료 선택)
    func calculate(srcAmount: String, srcMeterial: Meterial, dstTool: MeasuringTool) -> String

    // 인원 모드
    func calculate(srcPortion: String, srcAmount: String, srcUnit: MeasuringTool, dstPortion: String) -> String

    // 인원 모드 (재료 선택)
    func calculate(srcPortion: String, srcAmount: String, srcMeterial: Meterial, dstPortion: String) -> String

    // 재료 + 인원 모드
    func calculate(srcPortion: String, srcAmount: String, srcUnit: MeasuringTool, dstPortion: String, dstTool: MeasuringTool) -> String

    // 재료 + 인원 모드 (재료 선택)
    func calculate(srcPortion: String, srcAmount: String, srcMeterial: Meterial, dstPortion: String, dstTool: MeasuringTool) -> String
```
`calculator`의 위 함수를 사용하면 된다!

hihiboss committed:
iOS-#32 [Calculator] Refactor various getter functions of measuring tool list

hihiboss committed:
iOS-#32 [Calculator] Implement meterial model and manager

hihiboss committed:
iOS-#32 [Calculator] Implement calculator model

tngusmiso commented on Amoogye/AppDelegate.swift:
```diff
@@ -19,9 +19,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             UserDefaults.standard.set(false, forKey: SettingType.FirstExecution.rawValue)
         }
         setupDefaultMeasuringTool()
+        setupDefaultMeterial()
```
setupDefaultMeasuringTool()
setupDefaultMeterial()
이걸 if문 안이랑 밖에 두번이나 쓴 이유가 뭔가욤????

hihiboss replied:
@tngusmiso 
테스트하기 위함이었습니다!!
(한번 실행해버리면 다시 시뮬레이터 데이터 다 지우고 다시 실행해보고 해야해서...)

고쳐서 반영할게요!!

tngusmiso reviewed (dismissed):
확인했어욤~ 엄청나다

hihiboss committed:
iOS-#32 [Calculator] Apply review comments

- AppDelegate.swift의 setupDefaultMeasuringTool() 함수 호출 수정
- AppDelegate.swift의 setupDefaultMeterial() 함수 호출 수정

hihiboss commented:
@tngusmiso 
리뷰 반영했습니다! 확인부탁해용 :)

hihiboss commented:
#88 머지 반영 완료!

tngusmiso reviewed (approved):
👍 👍

hihiboss committed:
Merge branch 'develop' of https://github.com/Nexters/Amoogye-iOS into feature/calculator-model

- Fix the conflict in 'MeasuringToolViewController + ToolTabBar.swift'
